### PR TITLE
Updated travis and allow symfony 4

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,38 +1,59 @@
...
-php:
-    - 5.4
-    - 5.5
-    - 5.6
-    - 7.0
-    - 7.1
-    - nightly
+env:
+    global:
+        - PHPUNIT_FLAGS="-v"
```
Comment (stof) [inline .travis.yml]: As we always run PHPUnit in verbose mode, we could move this one outside the `PHPUNIT_FLAGS` (as done for `COMPOSER_FLAGS` containing only the job-specific flags)

File: .travis.yml [hunk 2]
```diff
@@ -1,38 +1,59 @@
...
+install:
+    # To be removed when this issue will be resolved: https://github.com/composer/composer/issues/5355
+    - if [[ "$COMPOSER_FLAGS" == *"--prefer-lowest"* ]]; then composer update --prefer-dist --no-interaction --prefer-stable --quiet; fi
+    - composer update ${COMPOSER_FLAGS} --prefer-dist --no-interaction
+    - ./vendor/bin/simple-phpunit install
 
-script: vendor/bin/simple-phpunit
+script:
+    - composer validate --strict --no-check-lock
+    - ./vendor/bin/simple-phpunit $PHPUNIT_FLAGS
```
Comment (stof) [inline .travis.yml]: missing final EOL

File: composer.json [hunk 3]
```diff
@@ -9,14 +9,14 @@
...
-        "symfony/console"          : "^2.7 || ^3.0",
+        "php"                      : "^5.4 || ^7.0",
+        "symfony/framework-bundle" : "^2.7 || ^3.0 || ^4.0",
+        "symfony/console"          : "^2.7 || ^3.0 || ^4.0",
         "henrikbjorn/stampie"      : "~0.11"
     },
     "require-dev": {
         "stof/stampie-extra": "~0.2@dev",
-        "symfony/phpunit-bridge": "^3.2"
+        "symfony/phpunit-bridge": "^3.4 || ^4.0"
```
Comment (stof) [inline composer.json]: you could just use `^4.0` here, as this bridge supports all PHP versions even in 4.x

**Comment by Nyholm** [comment]
Thank you for the review

File: composer.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
         "henrikbjorn/stampie"      : "~0.11"
     },
     "require-dev": {
-        "stof/stampie-extra": "~0.2@dev",
+        "stof/stampie-extra": "^1.0@beta",
```
Comment (fbourigault) [inline composer.json]: This should probably be in #23.

Comment (Nyholm) [reply composer.json]: You are correct. Let’s merge #23 first and I’ll rebase this later

File: .travis.yml [hunk 5]
```diff
@@ -1,34 +1,61 @@
...
+        - PHPUNIT_FLAGS=""
+        - SYMFONY_PHPUNIT_DIR="$HOME/symfony-bridge/.phpunit"
 
 matrix:
+    fast_finish: true
     include:
+          # Minimum supported Symfony version with the latest PHP version
+        - php: 7.2
+          env: COMPOSER_FLAGS="--prefer-stable --prefer-lowest" SYMFONY_DEPRECATIONS_HELPER="weak"
+        - php: 5.4
```
Comment (fbourigault) [inline .travis.yml]: PHP 5.4 has to be removed.

Comment (Nyholm) [reply .travis.yml]: Why?

Comment (fbourigault) [reply .travis.yml]: Stampie is 5.5+ now ;)

File: Tests/Functional/BundleInitializationTest.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        // Get the container
+        $container = $this->getContainer();
+
+        $this->assertTrue($container->has('hb_stampie.mailer.real'));
+        $service = $container->get('hb_stampie.mailer.real');
+        $this->assertInstanceOf(MailGun::class, $service);
+    }
+
+}
```
Comment (fbourigault) [inline Tests/Functional/BundleInitializationTest.php]: Missing EOL.

File: Tests/Functional/config.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+hb_stampie:
+    mailer: mailgun
+    server_token: foo:bar
+
+services:
+  httplug.client:
+    class: Http\Client\Curl\Client
```
Comment (fbourigault) [inline Tests/Functional/config.yml]: Missing EOL.

File: .travis.yml (hunk 5 above)
Comment (Nyholm) [reply .travis.yml]: Ah, okey =)

File: .editorconfig [hunk 8]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (fbourigault) [inline .editorconfig]: Missing EOL 🤣

Comment (Nyholm) [reply .editorconfig]: I thought that file would help me...

**Comment by Nyholm** [comment]
Wohoo, Travis is passing

**Pull request merged**